apps/dev-recruiters/src/components/modules/FilterPage/RolesFilterComponent/RolesFilterList/SearchRoles/index.tsx

```diff
@@ -10,29 +10,47 @@ import { Opportunity } from '@devlaunchers/models';
 interface Props {
   selectedRoleLabel: any;
   opportunities?: Opportunity[];
+  selectedRoleCategory?: String;
 }
-function SearchRole({ selectedRoleLabel, opportunities }: Props) {
+function SearchRole({
+  selectedRoleLabel,
+  opportunities,
+  selectedRoleCategory,
+}: Props) {
   const [selectedRole, setSelectedRole] = useState([]);
-
+  const [suggestedRole, setSuggestedRole] = useState([]);
   const { commitmentRange } = useOpportunitiesContext();
 
   useEffect(() => {
     if (!selectedRoleLabel) {
       const roleJsonString = localStorage.getItem('selectedRole');
       const selectedRole = JSON.parse(roleJsonString);
+      const selectedCategory = localStorage.getItem('roleCategory');
       setSelectedRole(selectedRole);
+      handleSuggestedRole(selectedCategory);
     } else if (commitmentRange !== null) {
       const filteredRoles = selectedRoleLabel.filter(
         (role) =>
-          role.commitmentHoursPerWeek >= commitmentRange.min &&
-          role.commitmentHoursPerWeek <= commitmentRange.max
+          role?.attributes?.commitmentHoursPerWeek >= commitmentRange.min &&
+          role?.attributes?.commitmentHoursPerWeek <= commitmentRange.max
       );
       setSelectedRole(filteredRoles);
+      handleSuggestedRole(selectedRoleCategory);
     } else {
       setSelectedRole(selectedRoleLabel);
+      handleSuggestedRole(selectedRoleCategory);
     }
   }, [selectedRoleLabel, commitmentRange]);
 
+  function handleSuggestedRole(RoleCategory) {

```

Comment:
Nit: should the variable name start with lower case `roleCategory`?

Reply:
Changes done.